### PR TITLE
 Fix anchor in case of heading which is only special characters

### PR DESCRIPTION
**Pull request by michiomochi**
Heading html's id parameter is empty character in case of only special characters.
If id parameter is empty character, the anchor doesn't work well so I fixed it.

## Before Behavior

Input Markdown

```
# ?
```

Output HTML

```
<h1>
<a id="" href="#" aria-hidden="true" class="anchor"><i class="fa fa-link"></i></a>?</h1>
```

## After Behavior

see test.
https://github.com/bitjourney/mato/compare/michiomochi/fix-heading?expand=1#diff-e89de987fbef031287e67c5536e8d07a

### CODE REVIEW

File: lib/mato/anchor_builder.rb [hunk 1]
```diff
@@ -44,7 +44,13 @@ def make_anchor_id_suffix(text)
     end
 
     def make_anchor_id_prefix(text)
-      ERB::Util.url_encode(text.downcase.gsub(/[^\p{Word}\- ]/u, "").tr(" ", "-"))
+      prefix = ERB::Util.url_encode(text.downcase.gsub(/[^\p{Word}\- ]/u, "").tr(" ", "-"))
+
+      if prefix.empty?
+        "user-content"
```
Comment (michiomochi) [inline lib/mato/anchor_builder.rb]: Assign `user-content` if anchor id is empty character.
`user-content` is used as anchor id prefix in GitHub, so use it.
mato a pipeline-based markdown toolkit with commonmark er 2019-01-11 19-52-21

File: test/html_filters/syntax_higlight_test.rb [hunk 2]
```diff
@@ -22,7 +22,10 @@ def test_guess_lexer
 
   def test_guess_lexer_for_ambiguous_filename
     # .pl is perl or prolog
-    assert { subject.guess_lexer(nil, 'foo.pl', '').tag == 'perl' }
+    assert do
+      tag = subject.guess_lexer(nil, 'foo.pl', '').tag
+      tag == "perl" || tag == "prolog"
```
Comment (michiomochi) [inline test/html_filters/syntax_higlight_test.rb]: `tag` is "prolog" in my environment, so changed it according to comment.

File: lib/mato/config.rb [hunk 3]
```diff
@@ -80,16 +80,19 @@ def configure(&block)
 
     def append_text_filter(text_filter, timeout: nil, on_timeout: nil, on_error: nil)
       raise "text_filter must respond to call()" unless text_filter.respond_to?(:call)
+
```
Comment (michiomochi) [inline lib/mato/config.rb]: Change by `rubocop -a`.

**Comment by michiomochi** [comment]
@bitjourney/engineers Please review 🙏

File: lib/mato/anchor_builder.rb (hunk 1 above)
Comment (gfx) [reply lib/mato/anchor_builder.rb]: 👍

**Comment by gfx** [comment]
LGTM 👏

File: lib/mato/anchor_builder.rb (hunk 1 above)
Comment (gfx) [reply lib/mato/anchor_builder.rb]: It's better if there's a comment as `# GitHub compatible`

Comment (michiomochi) [reply lib/mato/anchor_builder.rb]: Added
32d7266

**Comment by michiomochi** [comment]
Thanks!

**Pull request merged**